pkg/statuschecker/ready_checker.go

```diff
@@ -0,0 +1,168 @@
+package statuschecker
+
+import (
+	"fmt"
+
+	"github.com/atlassian/smith"
+	"github.com/atlassian/smith/pkg/resources"
+	"github.com/pkg/errors"
+	apiext_v1b1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type ObjectStatusType string
+
+const (
+	ObjectStatusTypeReady      ObjectStatusType = "Ready"
+	ObjectStatusTypeUnknown    ObjectStatusType = "Unknown"
+	ObjectStatusTypeInProgress ObjectStatusType = "InProgress"
+	ObjectStatusTypeError      ObjectStatusType = "Error"
+)
+
+type ObjectStatusResult interface {
+	StatusType() ObjectStatusType
+}
+
+type ObjectStatusReady struct {
+	Message string
```

Comment:
This is not used anywhere and also is not filled in everywhere (this is fine, but worth double checking if there are places where we could put something in there).